### ISSUE TITLE
sensu-go 5.6 changed behavior for listing with json format and no records

### ISSUE THREAD

**Issue by treydock** [sensu/sensu-go#2915]
## Expected Behavior

I'd expect `sensuctl <resource> list --format json` to continue returning empty array if no resources exist.

## Current Behavior

Starting with 5.6 the behavior changed and now returns `null` instead of empty array.  This is affecting most if not all resources.

## Steps to Reproduce (for bugs)

Ensure no checks exist:

```
[root@sensu-backend ~]# sensuctl check list --format json
null
```

BEFORE 5.6 on 5.5

```
[root@sensu-backend ~]# sensuctl check list --format json
[]
```

## Context

The puppet module for sensu assumed `--format json` would always return valid json and now that is not the case.

## Your Environment

* Sensu version used (sensuctl, sensu-backend, and/or sensu-agent): 5.6.0
* Installation method (packages, binaries, docker etc.): RPM
* Operating System and version (e.g. Ubuntu 14.04): CentOS 7

**Comment by treydock** [comment]
CC @ghoneycutt

**Comment by ghoneycutt** [comment]
Some quick searching shows that `null` is valid JSON. Does it not work with our JSON library?

**Comment by ccressent** [comment]
@treydock, I'm unable to reproduce this on 5.6:

```
$ sensuctl check list --format json
[]
```

```
$ sensu-backend version
sensu-backend version 5.6.0#4bd3afc, build 4bd3afcf64de3cb73bb12e9712c7e537a9a3144d, built 2019-05-08T18:40:14Z

$ sensuctl version
sensuctl version 5.6.0#4bd3afc, build 4bd3afcf64de3cb73bb12e9712c7e537a9a3144d, built 2019-05-08T18:40:58Z
```

Could you show me your output for `sensu-backend version` and `sensuctl version`?

**Comment by treydock** [comment]
Still an issue in my Vagrant instance and was an issue with automated Docker tests that start from fresh install.
```
[root@sensu-backend ~]# sensuctl version
sensuctl version 5.6.0#4bd3afc, build 4bd3afcf64de3cb73bb12e9712c7e537a9a3144d, built 2019-04-26T22:22:23Z
[root@sensu-backend ~]# sensu-backend version
sensu-backend version 5.6.0#4bd3afc, build 4bd3afcf64de3cb73bb12e9712c7e537a9a3144d, built 2019-04-26T22:20:54Z
[root@sensu-backend ~]# sensuctl check list --format json
null
```

At least for my Vagrant environment I deleted the etcd directory and restart sensu-backend and now everything is fine.  I don't know what could cause fresh etcd instance to change behavior but can't imagine how this would be a problem in our docker tests because they start with fresh install before the tests start adding and deleting resources.  For the docker tests I worked around issue in Ruby by turning nil to empty array as we had always assumed an array is returned.
